core/charm/charmpath.go

```diff
@@ -48,11 +48,7 @@ func NewCharmAtPathForceBase(path string, b base.Base, force bool) (charm.Charm,
 		}
 		return nil, nil, err
 	}
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return nil, nil, err
-	}
-	_, name := filepath.Split(absPath)
```

Comment:
Can you test what happens when you `refresh` a charm that was deployed before this fix and post this fix?

Reply:
Sure thing

Reply:
QA steps added